### PR TITLE
fix: WORKDAY.INTL won't overflow if all days are weekends

### PR DESCRIPTION
**Pull request by elsaminsut**
This PR addresses issue #623, where an edge case with a weekend mask of `1111111` would result in an overflow of the chrono crate. It adds an initial check for that weekend mask and results in a `#VALUE!` error if found. It also adds a small test to check the changes work as they should.

### CODE REVIEW

File: base/src/functions/date_and_time.rs [hunk 1]
```diff
@@ -1559,6 +1559,16 @@ impl Model {
             Ok(h) => h,
             Err(e) => return e,
         };
+
+        // Checks if all days are weekends
+        if weekend_mask == [true, true, true, true, true, true, true] {
```
Comment (nhatcher) [inline base/src/functions/date_and_time.rs]: What about `[true; 7]`?

**Review by nhatcher — CHANGES_REQUESTED** [review]
